docker-app/qfieldcloud/core/utils2/storage.py

```diff
@@ -19,6 +21,36 @@
 WEB_HTTPS_PORT = os.environ.get("WEB_HTTPS_PORT", None)
 
 
+def delete_by_prefix_versioned(prefix: str):
+    logging.info(f"S3 object deletion (versioned) with {prefix=}")
+
+    # prevent disastrous results when prefix is either empty string ("") or slash ("/"). Will delete random 1000 files.
+    if not isinstance(prefix, str) or prefix == "" or prefix == "/":
+        raise RuntimeError(f"Attempt to delete S3 object with illegal {prefix=}")
+
+    bucket = qfieldcloud.core.utils.get_s3_bucket()
+    return bucket.objects.filter(Prefix=prefix).delete()
+
+
+def delete_by_prefix_permanently(prefix: str):
+    logging.info(f"S3 object deletion (permanent) with {prefix=}")
+
+    # prevent disastrous results when prefix is either empty string ("") or slash ("/"). Will delete random 1000 file versions.
+    if not isinstance(prefix, str) or prefix == "" or prefix == "/":
+        raise RuntimeError(f"Attempt to delete S3 object with illegal {prefix=}")
+
+    bucket = qfieldcloud.core.utils.get_s3_bucket()
+    return bucket.object_versions.filter(Prefix=prefix).delete()
+
+
+def delete_version_permanently(version_obj: qfieldcloud.core.utils.S3ObjectVersion):
+    logging.info(
+        f'S3 object version deletion (permanent) with "{version_obj.key=}" and "{version_obj.id=}"'
+    )
+
+    version_obj._data.delete()
```

Comment:
(see comment below about keeping checks in one place, just as below since code style related, feel free to ignore as probably not worth debating this during the night :-) )